src/Sulu/Bundle/AdminBundle/Command/DownloadBuildCommand.php

```diff
@@ -147,22 +165,118 @@ protected function execute(InputInterface $input, OutputInterface $output)
         \unlink($tempFileZip);
     }
 
-    private function getLocaleFileHash(string $path)
+    private function getLocalFile(string $path)
     {
-        return $this->hash(\file_get_contents($this->projectDir . $path));
+        if (!\file_exists($this->projectDir . $path)) {
+            return '';
+        }
+
+        return \file_get_contents($this->projectDir . $path);
     }
 
-    private function getRemoteFileHash(string $path)
+    private function getRemoteFile(string $path)
     {
         $path = \str_replace(\DIRECTORY_SEPARATOR, '/', $path);
         $response = $this->httpClient->request('GET', $this->remoteRepository . $path);
 
-        return $this->hash($response->getContent());
+        return $response->getContent();
     }
 
     private function hash($content)
     {
         // we remove all whitespaces as the developer could change the indention or/and the line breaks of this files
         return \hash('sha256', \preg_replace('/\s+/', '', $content));
     }
+
+    private function writeFile(string $path, string $content): void
+    {
+        \file_put_contents($this->projectDir . $path, $content);
+    }
+
+    private function doManualBuild(SymfonyStyle $ui): int
+    {
+        $ui->warning(\sprintf(
+            'The files in the local "%s" folder do not match the ones in the remote repository "%s".' . \PHP_EOL
+            . 'Either bundles with custom JavaScript have been added, which means it has to be done manually '
+            . 'with NPM. Do you want start a manually build?',
+            static::ASSETS_DIR,
+            $this->remoteRepository
+        ));
+
+        $ui->title('Start manual build ...');
+
+        if ('y' !== \strtolower($ui->ask('Do you want start a manually build now?', 'y'))) {
+            return self::EXIT_CODE_INVALID_FILES;
+        }
+
+        $ui->section('Clean up node modules folders');
+        $this->cleanupNodeModulesFolders();
+
+        $ui->section('Install npm dependencies');
+        if ($this->runProcess($ui, 'npm install')) {
+            $ui->error('Could not build correctly install npm dependendencies.');
+
+            return self::EXIT_CODE_COULD_NOT_INSTALL_NPM_PACKAGES;
+        }
+
+        $ui->section('Build admin assets');
+        if ($this->runProcess($ui, 'npm run build')) {
+            $ui->error('Could not build correctly build admin assets.');
+
+            return self::EXIT_CODE_COULD_NOT_BUILD_ADMIN_ASSETS;
+        }
+
+        return 0;
+    }
+
+    private function cleanupNodeModulesFolders(): void
+    {
+        $removeBlockingFiles = [
+            'package-lock.json',
+            'yarn.lock',
+            'node_modules',
+        ];
+
+        $packageJson = \json_decode($this->getLocalFile(self::ASSETS_DIR . 'package.json'), true);
```

Comment:
i guess you dont want to remove any website assets of the project right? 
maybe we should just remove all `node_modules` directories and lockfiles inside of the `self::ASSETS_DIR` and the `vendor` folder?

Reply:
This function does only remove files in `assets/admin` and all it linked dependencies in `package.json`. All other package-lock and node_modules also in other vendors are untouched.

Reply:
yes i know, but you need to parse the `packag.json` to do that. maybe it would be easier to just remove these files inside of the `vendor` folder. do you see any problems with that?

Reply:
I'm afraid that then we are blamed that we did remove some `package-lock.json` which we should not remove thats why I did go with parsing the package.json and remove only the necessary files linked in that repository.